diff --git a/Thirdparty/DBoW2/DBoW2/FORB.cpp b/Thirdparty/DBoW2/DBoW2/FORB.cpp
index 1f1990c..80bf473 100644
--- a/Thirdparty/DBoW2/DBoW2/FORB.cpp
+++ b/Thirdparty/DBoW2/DBoW2/FORB.cpp
@@ -13,7 +13,7 @@
 #include <vector>
 #include <string>
 #include <sstream>
-#include <stdint-gcc.h>
+#include <stdint.h>

 #include "FORB.h"

diff --git a/Thirdparty/g2o/CMakeLists.txt b/Thirdparty/g2o/CMakeLists.txt
index 1a32ff9..1cb9b37 100644
--- a/Thirdparty/g2o/CMakeLists.txt
+++ b/Thirdparty/g2o/CMakeLists.txt
@@ -53,10 +53,10 @@ IF(OPENMP_FOUND AND G2O_USE_OPENMP)
   MESSAGE(STATUS "Compiling with OpenMP support")
 ENDIF(OPENMP_FOUND AND G2O_USE_OPENMP)

 # Compiler specific options for gcc
-SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -march=native")
-SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -march=native")
+SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
+SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3")
 # SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
 # SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3")

 # activate warnings !!!
diff --git a/src/LoopClosing.cc b/src/LoopClosing.cc
index 319afb5..9cedc76 100644
--- a/src/LoopClosing.cc
+++ b/src/LoopClosing.cc
@@ -981,9 +981,9 @@ void LoopClosing::CorrectLoop()
         cout << "Stoping Global Bundle Adjustment...";
         unique_lock<mutex> lock(mMutexGBA);
         mbStopGBA = true;

-        mnFullBAIdx++;
+        mnFullBAIdx = true; // Preserve the legacy C++11 bool++ behavior.

         if(mpThreadGBA)
         {
             mpThreadGBA->detach();
@@ -1232,9 +1232,9 @@ void LoopClosing::MergeLocal()
     {
         unique_lock<mutex> lock(mMutexGBA);
         mbStopGBA = true;

-        mnFullBAIdx++;
+        mnFullBAIdx = true; // Preserve the legacy C++11 bool++ behavior.

         if(mpThreadGBA)
         {
             mpThreadGBA->detach();
@@ -1805,9 +1805,9 @@ void LoopClosing::MergeLocal2()
     {
         unique_lock<mutex> lock(mMutexGBA);
         mbStopGBA = true;

-        mnFullBAIdx++;
+        mnFullBAIdx = true; // Preserve the legacy C++11 bool++ behavior.

         if(mpThreadGBA)
         {
             mpThreadGBA->detach();
diff --git a/src/ORBmatcher.cc b/src/ORBmatcher.cc
index 9129683..e26c678 100644
--- a/src/ORBmatcher.cc
+++ b/src/ORBmatcher.cc
@@ -24,9 +24,9 @@
 #include<opencv2/core/core.hpp>

 #include "Thirdparty/DBoW2/DBoW2/FeatureVector.h"

-#include<stdint-gcc.h>
+#include<stdint.h>

 using namespace std;

 namespace ORB_SLAM3
